packages/babel-plugin/src/index.tsx

```diff
@@ -11,6 +12,9 @@ export default declare<State>((api) => {
 
   return {
     inherits: require('babel-plugin-syntax-jsx'),
+    pre() {
+      this.cache = new Cache(this.opts);
```

Comment:
can this be placed in the module level? rationale is: this means we will only cache for the current file we're traversing. if we import the same module in another file - we will pay the cost again

Reply:
`pre()` should before babel start transforming. Check [this](https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#-pre-and-post-in-plugins). And if we import same module with same absolute path, it will bring the cached value. Moreover, I also wanted to set `this.opts` initially. I can pass `opts` on every function call but thought this looks better. What you say 🤔?

Reply:
> `pre()` should before babel start transforming right

per file unfortunately - not before the entire run